File: backend/schemas/review.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from datetime import datetime
from models.invoices import InvoiceStatus
from decimal import Decimal

class ReviewCreate(BaseModel):
    note: str | None = None

class ReviewerInfo(BaseModel):
    id: UUID
    full_name: str

    model_config = ConfigDict(from_attributes=True)

class ReviewResponse(BaseModel):
    invoice_id: UUID
    invoice_number: str
    status: InvoiceStatus
    reviewed_by: ReviewerInfo
    review_note: str | None = None
    reviewed_at: datetime

class FlaggedInvoiceItem(BaseModel):
    id: UUID
    invoice_number: str
    vendor_name: str
    grand_total: Decimal
    currency: str
    status: InvoiceStatus
    processed_at: datetime | None = None
    audit_report: str | None = None
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/review.ts
export type ReviewerInfo = {
  id: string
  full_name: string
}

export type FlaggedInvoiceItem = {
  id: string
  invoice_number: string
  vendor_name: string
  grand_total: number | string
  currency: string
  status: 'FLAGGED' | string
  processed_at: string | null
  audit_report: string | null
}

export type ResolveReviewPayload = {
  note: string | null
}

export type ResolveReviewResponse = {
  invoice_id: string
  invoice_number: string
  status: 'REVIEWED' | string
  reviewed_by: ReviewerInfo
  review_note: string | null
  reviewed_at: string
}

export type ReviewHistoryItem = {
  id: string
  invoice_id: string
  reviewed_by: string | ReviewerInfo
  review_note: string | null
  reviewed_at: string
}
